python/GafferUI/NumericWidget.py

```diff
@@ -75,7 +76,14 @@ def setValue( self, value ) :
 
 	def getValue( self ) :
 
-		return self.__numericType( self.getText() )
+		value = self.getText()
+
+		# Allow derived classes to modify the validator if required
+		validator = self._qtWidget().validator()
+		if isinstance( validator, BasicNumericMathExpression ) :
+			value = validator.eval( value )
```

Comment:
As I understand it, the only reason this new value makes it into the final text is because `__editingFinished()` calls `getValue()` and then does a reformatting. At a minimum, I think we need a comment in `__editingFinished()` to say that this is how expressions get evaluated. But perhaps it might be clearer to do the eval only in `__editingFinished()`? Presumably that would be a problem if someone called `getValue()` directly while there was an expression in there, but we already have an equivalent problem : what should we return when someone calls `getValue()` and there is an incomplete expression, such as `1/`?

Reply:
I've left it in `getValue` as now we [validate incomplete expressions](https://github.com/GafferHQ/gaffer/pull/3873/commits/4505a343edc81a5073011877110c4b503c8b1efa), we should be in a much better place to always return something sensible. It felt more robust that the expansion would happen when the value was queried without relying on UI events. Happy to change if you'd rather though. Comment added [here](https://github.com/GafferHQ/gaffer/pull/3873/commits/dc49f528883fcf7610c79d042a248453f4027fcb).